### ISSUE TITLE
Mismatched labels in visualize_hierarchy

### ISSUE THREAD

**Issue by jalkestrup** [MaartenGr/BERTopic#1207]
Hi Maarten

Thanks for a super great tool and all the hard work that have clearly been put into BERTopic.  I am currently working with data from the Danish parliament, but have some unexpected behavior with the labels of the visualize_hierarchy function. Perhaps I am missing something, but it seems like the Topics shown on the left side of the visualizing are shifted - they don't make sense in combination with the topics to which they are merged:

Example, on the left top side I have these 2 topics (screenshot 1)
44_oliefyr_energinet_varme
33_energiaftalen_energinet

![screenshot 1](https://user-images.githubusercontent.com/17813449/233863368-05004218-687e-47e1-988b-4b4325141701.png)

But the supposed merge (screenshot 2) is per_clausen_per_skat_regning - which makes no sense. However, if I instead hover over the horisontal line next to the topics I believe I see what is the correct topics, fx. skatten_afgiften_skatter_no_skatteministeren (screenshot 1).

![screenshot 2](https://user-images.githubusercontent.com/17813449/233863371-3c6233b2-a98d-4ca7-a136-3fb5bf581635.png)

The topics seem to be correct if I create the tree and print that (hierachy.txt), so it just seem to be a matter of the labels on the y-axis of the visualize_hierarchy that have been shifted.

[hierachy.txt](https://github.com/MaartenGr/BERTopic/files/11304823/hierachy.txt)

```
sbert_model = SentenceTransformer("paraphrase-multilingual-MiniLM-L12-v2", device='cpu')
q_sbert_model = torch.quantization.quantize_dynamic(sbert_model, {torch.nn.Linear}, dtype=torch.qint8)
q_clim_statement_emb= q_sbert_model.encode(statements_climate, show_progress_bar=True)

from sklearn.feature_extraction.text import CountVectorizer
from hdbscan import HDBSCAN

hdbscan_model_cs1 = HDBSCAN(min_cluster_size=15, metric='euclidean', prediction_data=True)
umap_model_cs1 = UMAP(n_neighbors=10, n_components=5, min_dist=0.0, metric='cosine', angular_rp_forest=True, random_state=42)
vectorizer_model_cs1 = CountVectorizer(ngram_range=(1, 3), stop_words=dk_stop_words, encoding='utf-8', min_df=10, max_df=0.5)
topic_model_cs1 = BERTopic(calculate_probabilities=False, umap_model=umap_model_cs1, low_memory=True, hdbscan_model=hdbscan_model_cs1, min_topic_size=30, vectorizer_model=vectorizer_model_cs1, verbose=True, language=None)
topic_cs1,_ = topic_model_cs1.fit_transform(statements_climate, q_clim_statement_emb)

hierarchical_topics_cs1 = topic_model_cs1.hierarchical_topics(statements_climate)
topic_model_cs1.visualize_hierarchy(hierarchical_topics=hierarchical_topics_cs1)
```

(FYI I have also tried with the regular sbert not quantized just to check, and it makes no difference)

Thanks a lot for the help

Kind regards,
Jesper

**Comment by jalkestrup** [comment]
By the way, I am running BERTopic 0.14.0 in Python 3.10, attached is my conda env file

[nlp_conda_environment.txt](https://github.com/MaartenGr/BERTopic/files/11304868/nlp_conda_environment.txt)

**Comment by MaartenGr** [comment]
Apologies for the late reply! This is a known issue and was fixed in the main branch of this repository. If you install BERTopic from the main branch the issue should be fixed.

**Comment by MaartenGr** [comment]
Closing this due to inactivity. Let me know if I need to re-open the issue!
